providers/transip/transipProvider.go

```diff
@@ -109,47 +108,142 @@ func (n *transipProvider) GetDomainCorrections(dc *models.DomainConfig) ([]*mode
 
 	models.PostProcessRecords(curRecords)
 
+	if !diff2.EnableDiff2 {
+		corrections, err := n.getCorrectionsUsingOldDiff(dc, curRecords)
+		return corrections, err
+	}
+
+	corrections, err := n.getCorrectionsUsingDiff2(dc, curRecords)
+	return corrections, err
+}
+
+func (n *transipProvider) getCorrectionsUsingDiff2(dc *models.DomainConfig, records models.Records) ([]*models.Correction, error) {
 	var corrections []*models.Correction
-	if !diff2.EnableDiff2 || true { // Remove "|| true" when diff2 version arrives
+	instructions, err := diff2.ByRecordSet(records, dc, nil)
+	if err != nil {
+		return nil, err
+	}
 
-		differ := diff.New(dc)
-		_, create, del, modify, err := differ.IncrementalDiff(curRecords)
-		if err != nil {
-			return nil, err
+	for _, change := range instructions {
+
+		if canDirectApplyDNSEntries(change) {
+			changeFunction := n.getChangeFunction(change.Type, dc, change)
+			corrections = append(corrections, change.CreateCorrection(changeFunction))
+		} else {
+			deleteFunction := n.getChangeFunction(diff2.DELETE, dc, change)
+			createFunction := n.getChangeFunction(diff2.CREATE, dc, change)
+			corrections = append(corrections, change.CreateCorrectionWithMessage("[1/2] delete", deleteFunction), change.CreateCorrectionWithMessage("[2/2] create", createFunction))
 		}
+	}
+
+	return corrections, nil
+}
+
+func (n *transipProvider) getChangeFunction(changeType diff2.Verb, dc *models.DomainConfig, change diff2.Change) func() error {
```

Comment:
Is there really a benefit to having getChangeFunction?  I assume it is used to be DRY, but without it there wouldn't be much repetition: The case statements below would be copied to getCorrectionsUsingDiff2 once; the `else` condition would repeat the DELETE and CREATE cases, which are 1 line each.

The benefit would be code that is easier to understand for the next maintainer.

I'm not saying you have to make this change, but do consider it.

Reply:
Yeah this comes down to preference I think. I'd like to have my functions short-lived and shorter in line length. 

I inlined the function as requested this made adding the diff2.REPORT also allot more manageable.